### PR TITLE
[Config] Change default barriers of mainnet to 0

### PR DESCRIPTION
**Pull request by smartprogrammer93**
## Changes

- change default of barriers to 0 for mainnet without affecting previously synced nodes

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

- @shashankshampi we need to run a test for this one where we sync a node with default configurations from master then update the node to this branch and check if it syncs the old data or not. if not, then all is good.

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

default to barriers on mainnet configuration changed to 0 to support the network. If you wish to use the old default barrier you need to set it manually.

### CODE REVIEW

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:44]: are you sure it will work correctly in every case?

File: src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs:235]: tests for ReceiptsSyncFeed, but no tests for bodies
no tests for every possible case

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs:235]: There are no tests for bodies in the first place. which was surprising tbh

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:44]: We are covering 99 percent of cases. the only case that is not covered is if someone is syncing, and suddenly stopped the client at exactly around the old default, then unfortunately the sync will stop. other than that, it is all in there

File: src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs]: Added more cases for receipts

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg
Comment (deffrian) [inline src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:20]: Do we want to remove these values completely? If user decide to enable them he can set value close to the head and CL will not be able to sync deposit contract. Maybe we can introduce something like `Sync.EnableBarrier`?

Comment (deffrian) [reply src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:20]: Fair, but I think it's better option to have default value for this. User may decide to set it to current head

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:20]: If that is desired, we can just make another config file, like `mainet-barriers`, for example

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs
Comment (asdacap) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:44]: Do other network have barrier too?

**Comment by asdacap** [comment]
You'll need to change `SyncProgressResolver` also, there is a duplicated logic there. See https://github.com/NethermindEth/nethermind/pull/6184 for an attempt to consolidate the logic.

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:44]: Yeah, they do. The assumption is we only want to do that for mainnet

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:44]: By default we only had barriers on mainnet

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg
Comment (kamilchodola) [reply src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:20]: I'd better go with `mainnet-light` or something but not sure.

**Comment by smartprogrammer93** [comment]
Following tests were perfomed
- Sync this branch fully on mainnet (all bodies and receipts synced)
- Sync Master with barriers and then switch to this branch
Dont see any issues.

The current implementation covers the following case:
- Node syncing from scretch = syncing all bodies and receipts
- Node already synced with barriers = no change. It will not sync the rest even with the config change.
- Node syncing from scratch restarted at any point, including at the previous barrier default = will continue sync no problem.
- Node syncing from scratch with barriers configured = will stop at the barrier no problem
I believe this covers all cases.

**Review by asdacap — APPROVED** [review]

**Comment by kamilchodola** [comment]
@smartprogrammer93 Let's make sure it is well tested since may break some setups.

@shashankshampi Important topic - please prepare a lot of scenarios including nodes synced on all version (before and after autopivot). Please also put a summary of testing here under PR before it would be merged - let's don't rush this one - it would be included in next version so we have a lot of time to make sure we are well prepared.

**Comment by shashankshampi** [comment]
The test is done and looks good to go.

**Review by shashankshampi — APPROVED** [review]
Tested this branch with couple of scenarios and looks stable to process ahead

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs:106]: Edge case (not sure if worth handling): if the mainnet user with config barriers different than default deposits will upgrade Nethermind during receipts download it will stop receipts sync.

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs:95]: Think about refactoring BodiesSyncFeed and ReceiptSyncFeed to reuse the same (similar) code

**Review by MarekM25 — APPROVED** [review]
It looks much better now. Added two comments to consider, but PR looks good

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs:106]: I would ignore case when someone upgrades during sync

**Comment by smartprogrammer93** [comment]
Retested and got green light from kamil. Merging

**Pull request merged**
